src/components/views/articles/SingleArticle.js

```diff
@@ -7,19 +9,24 @@ import { connect } from "react-redux";
 import PropTypes from "prop-types";
 import { bindActionCreators } from "redux";
 import { Link } from "react-router-dom";
-
+import { Button } from "semantic-ui-react";
 import * as actionGenerators from "../../../actions/articlesActions";
+import * as bookmarkActions from "../../../actions/bookmarkArticleActions";
 import HeaderComponent from "../../containers/headers/index";
+import "../../../assets/style/articles/bookmark.scss";
 
 class SingleArticle extends Component {
   constructor(props) {
     super(props);
     this.state = {
       userIsAuthor: false,
-      currentArticle: {}
+      currentArticle: {},
+      bookmarked: props.bookmarked
     };
 
     this.checkIfAuthenticated = this.checkIfAuthenticated.bind(this);
+    this.bookmarkArticle = this.bookmarkArticle.bind(this);
+    this.unBookmarkArticle = this.unBookmarkArticle.bind(this);
   }
 
   componentDidMount = () => {
```

Comment:
why do we have this as an arrow function?

Reply:
Fixed it.